src/api/ledger/parse/trustline.js

```diff
@@ -4,19 +4,30 @@ const assert = require('assert');
 const utils = require('./utils');
 const flags = utils.core.Transaction.flags.TrustSet;
 
+function parseFlag(flagsValue, trueValue, falseValue) {
+  if (flagsValue & trueValue) {
+    return true;
+  }
+  if (flagsValue & falseValue) {
+    return false;
+  }
+  return undefined;
+}
+
 function parseTrustline(tx: Object): Object {
   assert(tx.TransactionType === 'TrustSet');
 
-  return {
+  return utils.removeUndefined({
     limit: tx.LimitAmount.value,
     currency: tx.LimitAmount.currency,
     counterparty: tx.LimitAmount.issuer,
     qualityIn: tx.QualityIn,
     qualityOut: tx.QualityOut,
-    disableRippling: (tx.Flags & flags.NoRipple) !== 0,
-    frozen: (tx.Flags & flags.SetFreeze) !== 0,
-    authorized: (tx.Flags & flags.SetAuth) !== 0
-  };
+    ripplingDisabled: parseFlag(
+      tx.Flags, flags.SetNoRipple, flags.ClearNoRipple),
+    frozen: parseFlag(tx.Flags, flags.SetFreeze, flags.ClearFreeze),
+    authorized: parseFlag(tx.Flags, flags.SetAuth, 0)
```

Comment:
Yeah, it's probably intentional. Once you authorize a trustline, it would be awkward to allow revoking it because it might have a balance on it at that point.